service/storage_service.cc

```diff
@@ -5984,7 +5984,21 @@ future<> storage_service::clone_locally_tablet_storage(locator::global_tablet_id
 // Streams data to the pending tablet replica of a given tablet on this node.
 // The source tablet replica is determined from the current transition info of the tablet.
 future<> storage_service::stream_tablet(locator::global_tablet_id tablet) {
-    return do_tablet_operation(tablet, "Streaming", [this, tablet] (locator::tablet_metadata_guard& guard) -> future<> {
+    co_await utils::get_local_injector().inject("block_base_tablet_streaming", [this, &tablet] (auto& handler) -> future<> {
+        if (!_db.local().find_column_family(tablet.table).schema()->is_view()) {
+            while (!handler.poll_for_message() && !_async_gate.is_closed()) {
+                co_await sleep(std::chrono::milliseconds(100));
+            }
+        }
+    });
+    co_await utils::get_local_injector().inject("block_view_tablet_streaming", [this, &tablet] (auto& handler) -> future<> {
+        if (_db.local().find_column_family(tablet.table).schema()->is_view()) {
+            while (!handler.poll_for_message() && !_async_gate.is_closed()) {
+                co_await sleep(std::chrono::milliseconds(100));
+            }
+        }
+    });
```

Comment:
Can error injections be parameterized? If so, we wouldn't have to pretty much duplicate the code here.

CC @kbr-scylla I'm under the impression that these error injections are slowly getting out of control and litter the code. It't not like you're responsible for this, just sharing an impression.

Reply:
> Can error injections be parameterized? 

Yes

Reply:
Ok, I replaced the error injections with one that's parametrized